### ISSUE TITLE
Creating Fat Jar File failed for Unsupported JavaFX Runtime

### ISSUE THREAD

**Issue by leawang0** [nus-cs2103-AY2324S1/forum#141]
I am attempting to create a fat jar file including the JavaFX in IntelliJ (windows). I can launch my chatbot and its GUI without any problem using gradle run.
However, when I run `./gradlew shadowJar`. Such an error occurs:

```
FAILURE: Build failed with an exception.

* What went wrong:
java.lang.UnsupportedClassVersionError: org/openjfx/gradle/JavaFXPlugin has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Ja
va Runtime only recognizes class file versions up to 52.0
> org/openjfx/gradle/JavaFXPlugin has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file 
versions up to 52.0

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 10s
```

I am unable to find a good solution to this online, and previously when I was doing GUI, I have already configured the project JDK and Gradle JDK to be the same Java 11. Does anyone face a similar issue or know how to resolve this?

**Comment by jx124** [comment]
This might be caused by a misconfiguration in your `build.gradle` file. Perhaps the version numbers used are incompatible? Here is my `build.gradle` file for your reference:

```
plugins {
    id 'java'
    id 'application'
    id 'com.github.johnrengelman.shadow' version '7.1.2'
    id 'org.openjfx.javafxplugin' version '0.0.13'
}

repositories {
    mavenCentral()
}

javafx {
    version = '17'
    modules = [ 'javafx.controls', 'javafx.fxml' ]
}

dependencies {
    testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: '5.10.0'
    testRuntimeOnly group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: '5.10.0'

    String javaFxVersion = '17.0.7'

    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-base', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-controls', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-fxml', version: javaFxVersion, classifier: 'linux'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'win'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'mac'
    implementation group: 'org.openjfx', name: 'javafx-graphics', version: javaFxVersion, classifier: 'linux'
}

test {
    useJUnitPlatform()

    testLogging {
        events "passed", "skipped", "failed"

        showExceptions true
        exceptionFormat "full"
        showCauses true
        showStackTraces true
        showStandardStreams = false
    }
}

application {
    mainClass.set("duke.Launcher")
}

shadowJar {
    archiveBaseName = "duke"
    archiveClassifier = null
    dependsOn("distZip", "distTar")
}

run{
    standardInput = System.in
}
```

**Comment by damithc** [comment]
@wanghejin In the same terminal you ran the gradle command, also run the `java -version` command to see which Java version is being used as the default.

**Comment by leawang0** [comment]
Thank you for the suggestions! I updated my `build.gradle` to be the same as @jx124 but realised it still doesn't work because of java issues. Then I uninstalled the current version of Java and reinstalled again. I also reset my environment variables, project sdk and gradle sdk to point to the newly installed version of Java. Then things went well.

Thank you @jx124 and prof for taking the time to answer my queries!
